connaisseur/validators/notaryv1/trust_data.py

```diff
@@ -93,11 +94,28 @@ def validate_signature(self, keystore: KeyStore):
             sig = signature["sig"]
 
             try:
-                verify_signature(pub_key, sig, msg)
+                TrustData.__validate_signature_with_key(sig, msg, pub_key)
             except Exception as err:
                 msg = "Failed to verify signature of trust data {trust_data_kind}."
                 raise ValidationError(message=msg, trust_data_kind=self.kind) from err
 
+    @staticmethod
+    def __validate_signature_with_key(signature: str, payload: str, key: Key):
+        if isinstance(key, ECDSAKey):
+            return TrustData.__validate_signature_with_ecdsa(signature, payload, key)
+        msg = (
```

Comment:
Given that this function can already render the error string, is there a reason to delegate rendering the error string to the error class?

Reply:
i don't understand ...?

Reply:
"The key type {key_type} is unsupported for a validator of type {val_type}." is not rendered as the error string, but rather is the template. The function, where I commented does provide the template and the data inserted into the template. Why let the error class handle rendering the error message instead of simply providing the error that this function apparently knows it wants to output?

Reply:
created issue #646